src/java/com/zimbra/clam/ClamScanner.java

```diff
@@ -102,73 +118,83 @@ protected Result accept(InputStream is, StringBuffer info) {
         }
     }
 
-    private static final byte[] lineSeparator = { '\r', '\n' };
-
-    private Result accept0(byte[] data, InputStream is, StringBuffer info) throws UnknownHostException, IOException {
-        Socket commandSocket = null;
-        Socket dataSocket = null;
-
+    private Result accept0(byte[] data, InputStream inputStream, StringBuffer info) throws UnknownHostException, IOException {
+        Socket socket = null;
+        OutputStream outStream = null;
+        InputStream socketInputStream = null;
         try {
-            if (LOG.isDebugEnabled()) { LOG.debug("connecting to " + mClamdHost + ":" + mClamdPort); }
-            commandSocket = new Socket(mClamdHost, mClamdPort);
-
-            BufferedOutputStream out = new BufferedOutputStream(commandSocket.getOutputStream());
-            TcpServerInputStream in = new TcpServerInputStream(commandSocket.getInputStream());
-
-            if (LOG.isDebugEnabled()) { LOG.debug("writing STREAM command"); }
-            out.write("STREAM".getBytes("iso-8859-1"));
-            out.write(lineSeparator);
-            out.flush();
-
-            if (LOG.isDebugEnabled()) { LOG.debug("reading PORT"); }
-            // REMIND - should have timeout's on this...
-            String portLine = in.readLine();
-            if (portLine == null) {
-                throw new ProtocolException("EOF from clamd when looking for PORT repsonse");
+            if (LOG.isDebugEnabled()) {
```

Comment:
we could remove this condition of logging as we are anyways logging to debug level

Reply:
I am not sure but seems these checks were to make sure the log statement gets printed only when extensions logging set to debug.

Reply:
since the logging level is debug it wont print to logs unless log level for extensions is set to debug/trace, could give it a try and check

Reply:
Ok, here is the reason which I found. Usually, these checks are added for performance improvement.
Logger checks internally for log level, but first, it evaluates the string in the msg parameter before calling logging methods(`+` concatenation or String format these are expensive operations). 
see this: https://stackoverflow.com/questions/6504407/is-there-a-need-to-do-a-iflog-isdebugenabled-check

I had never used these debug checks before but it seems beneficial for the high-traffic code section, I think we should follow this whenever required.